app/controllers/users_controller.rb

```diff
@@ -17,13 +17,17 @@ def create
 
   def edit
     @user = User.find(params[:id])
+    render file: "public/404" unless @user == current_user
```

Comment:
why do we need to render 404 here? This is already done as a before action in the ApplicationController